Automattic/wp-calypso
# Highlight Business plan as "Popular"

#### Changes proposed in this Pull Request

Check the Marketing Testing post on p8Eqe3-Zs-p2.

* This PR introduces a new A/B test that highlights the Business plan with a "Popular" badge in the signup flow. We are also removing an old test `popularPlanBy` and references to that test introduced in https://github.com/Automattic/wp-calypso/pull/33291.
* Presently, choosing site type as either `Business` or `Online Store` already highlights the Business plan as "Popular". This A/B test is only when the selected site type is `Blog` or `Professional`.
* The plans page at `/plans` and launch flow are excluded from the test.

<img width="1666" alt="Screenshot 2019-12-11 at 9 32 23 PM" src="https://user-images.githubusercontent.com/1269602/70637751-c2906280-1c5d-11ea-9c0f-c5ad117e8a06.png">


#### Testing instructions



**Scenario 1**(_Verify current behaviour remains unchanged_)

1. Go through the signup flow and select either `Business` or `Online Store` as site type. 
2. Verify in Plans step that the Business plan is highlighted as Popular.

**Scenario 2**(_Business plan should show as "Popular" only in signup - /plans page and launch flow are unaffected_)

_UPDATE_: _We've updated the test to non-en and non-US(check Scenario 4 below) so ensure you satisfy these conditions before testing this scenario_

1. Put yourself in `variantShowBizPopular` variation of the `showBusinessPlanPopular` test and begin the signup flow.
2. In the site type step, select either `Blog` or `Professional`
3. Verify that the Business plan is highlighted as Popular in the plans step.
4. Complete signup with a free plan, and head to the plans page at /plans. Verify that on the `Blogs and Personal sites` tab, the Premium plan is shown as Popular, and on the `Business sites and Online Stores` tab, the Business plan is shown as popular. That is, the test should not affect /plans page.
5. Go through the launch flow and verify that Premium plan is shown as popular. That is, the test should not affect the launch flow.

**Scenario 3** (_Verify test assignment does not happen when Plans page is viewed outside of signup flow_)

1. Go through the signup flow and and on the domains step, open your browser console and type `localstorage.ABTests;`. The `showBusinessPlanPopular` should not show i.e at any point before the plans step, no test assignment should happen.
2. Login to an existing wpcom account that has a private site. Go to /plans page. Then, go through the launch flow and launch the site. Open your browser console and type `localstorage.ABTests;`. The `showBusinessPlanPopular` should not show.
3. Go through the signup flow and in the site type step, select either `Business` or `Online Store`. Then complete signup. Open your browser console and type `localstorage.ABTests;`. The `showBusinessPlanPopular` should not show(since you will be assigned to the test only if site type is Blog or Professional)

**Scenario 4**

1. Go through the signup flow and verify that you are _not_ assigned to the test if your country code is US. If you are not in the US, simulate your location by typing the following in your browser console:
```
// in browser console, set country code to US. 
window.userCountryCode = 'US';
```
2. Go through the signup flow in `en` locale and verify that you are _not_ assigned to the test if your country code is non-US. If you are in the US, simulate your location by typing the following in your browser :
```
// in browser console, type a non-US country code
window.userCountryCode = 'IN';
```

3. Go through the signup flow in any non-en locale - for example`http://calypso.localhost:3000/fr`- and any non-US country - simulate your location by typing  `window.userCountryCode = 'IN'; `in your browser console - and verify that you are assigned to the test in the plans step of the signup flow.


You can also try the steps above by using a proxy service like TunnelBear(which is free to use) to simulate the country code.
